app/services/cancel_comment.rb

```diff
@@ -0,0 +1,13 @@
+class CancelComment
```

Comment:
Directly calls `comment.destroy` from the controller.

**N.b.** My bad on this one. You certainly took inspiration from my services, but those were just skeletons before the action/notifications patch.